### PR TITLE
Add parenthesis around 'in' operator

### LINKED ISSUE #13: 'in'-Operator needs to be grouped together
When using the `in`-Operator with following filter the query doesn't reflect the implied `and`
```
filter: {
        bar: 1,
        foo: {
            in: ['a', 'b'],
            contains: 'foo'
        }
    }
}
```

### Result
`?$filter=bar eq 1 and foo eq 'a' or foo eq 'b' and contains(foo,'foo')`

### Expected Outcome
`?$filter=bar eq 1 and (foo eq 'a' or foo eq 'b' and contains(foo,'foo'))`

### PR DESCRIPTION
**Pull request by samuelportz**
This fixes #13, but an in operator whit an implied and will result in double paranthesis

```
[{
    SomeProp: {
         in: [1, 2, 3]
    }
},
{
    AnotherProp: 4
}];
```

```
?$filter=((SomeProp eq 1 or SomeProp eq 2 or SomeProp eq 3)) and (AnotherProp eq 4)
```

### CODE REVIEW

**Comment by techniq** [comment]
I've always relied on the use of Arrays to drive where to place parens but I could be talked into this as it's a fairly common occurrence to have to handle.

Before this PR, you're example above...
```js
const filter = [{
  SomeProp: {
    in: [1, 2, 3]
  }
},
{
  AnotherProp: 4
}];

buildFilter({ filter })
```
would produce...
```
?$filter=(SomeProp eq 1 or SomeProp eq 2 or SomeProp eq 3) and (AnotherProp eq 4)
```

Doing the same with your example in #13...
```js
const filter = [{
  bar: 1
}, {
  foo: {
    in: [1, 2, 3],
    contains: 'foo'
  }
}];

buildFilter({ filter })
```
produces...
```
"?$filter=(bar eq 1) and (foo eq 'a' or foo eq 'b' and contains(foo,'foo'))"
```
which matches your expected outcome per the issue (other than also wrapping `bar eq 1`, which has no net difference), although I figure you probably want
```
"?$filter=(bar eq 1) and ((foo eq 'a' or foo eq 'b') and contains(foo,'foo'))"
```
This example (which is contrived I'm sure) doesn't make a lot of sense as `foo` couldn't equal 'a' or 'b' and still contain 'foo', but I get the idea.

In short, if you could verify the last output is supported by this PR, I'm willing to merge it.  I just want to make sure it doesn't hinder the ability to express other filters.

**Comment by samuelportz** [comment]
Hi!
I just realized that I totally (unintetionally) ignored the possibility to wrap my statements in arrays. 
I want to add how I came to this pr.

I allow a user to search in a specific field at the frontend. Then I add in the backend the list of allowed values which can vary between different users.
My result will something like 
```js
const filter = {
    SomeNames: {
        contains: "Bob", // Fulltext search
        in: ["Peter Newman", "Bob Ross", "Bobby Parker", "Mike Bobson"], // Allowed values
    }
};
```
which would result in 
```
?$filter=contains(SomeNames,'Bob') and SomeNames eq 'Peter Newman' or SomeNames eq 'Bob Ross' or SomeNames eq 'Bobby Parker' or SomeNames eq 'Mike Bobson'
```
and have no results

Correct would be
```
?$filter=contains(SomeNames,'Bob') and (SomeNames eq 'Peter Newman' or SomeNames eq 'Bob Ross' or SomeNames eq 'Bobby Parker' or SomeNames eq 'Mike Bobson')
```
which will have 3 results.

I just changed the test examples (also I saw that I duplicated the in-test). 

Addtional parenthesis around operators won't alter the logic so the last output is also supported. Wofram alpha was very helpful as I tried to remember boolean algebra^^
As you can see the minimal forms for the last three examples are the same regardless the additional parenthesis.
https://www.wolframalpha.com/input/?i=(a)+and+(b+or+c+and+d)
https://www.wolframalpha.com/input/?i=(a)+and+(b+or+c)+and+d
https://www.wolframalpha.com/input/?i=(a)+and+((b+or+c)+and+d)
https://www.wolframalpha.com/input/?i=(a)+and+(((b+or+c))+and+d)

**Comment by techniq** [comment]
Thanks for the contribution.  Released as version `5.2.0`.

I made a minor change to one of your tests.  Instead of:
```js
const filter = [{ SomeProp: { in: [1, 2, 3] } }, { AnotherProp: 4 }];
const expected = '?$filter=((SomeProp eq 1 or SomeProp eq 2 or SomeProp eq 3)) and (AnotherProp eq 4)'
```

I changed it to
```js
const filter = [{ SomeProp: { in: [1, 2, 3] }, AnotherProp: 4 }];
const expected = '?$filter=((SomeProp eq 1 or SomeProp eq 2 or SomeProp eq 3) and AnotherProp eq 4)'
```

as putting `{ AnotherProp: 4 }` into its own object causes it to be wrapped in parens separately.

The way arrays are currently handled is they wrap each item in their own parens.  Maybe instead of ([code](https://github.com/techniq/odata-query/blob/master/src/index.js#L96-L100))
```js
} else if (Array.isArray(filters)) {
    const builtFilters = filters.map(f => buildFilter(f, propPrefix)).filter(f => f !== undefined);
    if (builtFilters.length) {
      return `${builtFilters.map(f => `(${f})`).join(` and `)}`
    }
```
it should be:
```js
} else if (Array.isArray(filters)) {
    const builtFilters = filters.map(f => buildFilter(f, propPrefix)).filter(f => f !== undefined);
    if (builtFilters.length) {
      return `(${builtFilters.join(` and `)})`
    }
```
so the `[ ]` would literally translate as `( )` in the query.  If this change is made, I would release it as a new major version since it could be breaking for some users.

Thanks again for your contribution.
